packages/transformer-directives/src/apply.ts

```diff
@@ -83,7 +84,8 @@ export async function parseApply({ code, uno, applyVariable }: TransformerDirect
     if (parent || (selectorOrGroup && selectorOrGroup !== '.\\-')) {
       let newSelector = generate(node.prelude)
       const className = code.slice(node.prelude.loc!.start.offset, node.prelude.loc!.end.offset)
-      if (selectorOrGroup && selectorOrGroup !== '.\\-') {
+      // if mode is 'vue-scoped', we don't need to scope the selector
+      if ((uno.userConfig as VitePluginConfig).mode !== 'vue-scoped' && selectorOrGroup && selectorOrGroup !== '.\\-') {
```

Comment:
This `mode` property is not on `userConfig`

Reply:
It seems that there are some problems with the type, but it does exist at runtime. It should be merged by the userConfig from vitePlugin.

Reply:
https://github.com/unocss/unocss/blob/243623ddb7decd7002e4110ef3674e1feeef6297/packages/config/src/index.ts#L59-L60

It seems that when loading the config, the vite plugin config is not merged into the `userConfig`

Reply:
https://github.com/unocss/unocss/blob/main/packages/vite/src/index.ts#L35
Go down from here and there is a merge operation

<img width="934" alt="image" src="https://github.com/user-attachments/assets/bd488dac-4dab-4f8a-b695-e51c8f9a1fda">

<img width="978" alt="image" src="https://github.com/user-attachments/assets/40af0177-16ab-4972-bb8d-a8a856e5b244">

Reply:
<img width="646" alt="image" src="https://github.com/user-attachments/assets/d9599e11-1181-4d01-9e6b-e29114e0a952">
<img width="706" alt="image" src="https://github.com/user-attachments/assets/16f81687-0128-4687-ad8b-412fb86fe1b0">

vite plugin config is **not** injected into the `context`.

Reply:
https://github.com/unocss/unocss/blob/main/packages/vite/src/index.ts#L48
`ConfigHMRPlugin` will trigger reloadConfig and then inject
<img width="1003" alt="image" src="https://github.com/user-attachments/assets/d6913082-79fe-4fd1-96ab-3bb28ec0714d">

Reply:
I think `reloadConfig` still won't inject vite config in `userConfig`, it is a configuration item of `unocss` not `vite` plugin option. 🧐

Reply:
<img width="967" alt="image" src="https://github.com/user-attachments/assets/aad46596-7e8f-4db0-87d3-ed91f210a6a2">

<img width="1469" alt="image" src="https://github.com/user-attachments/assets/136b4f8d-bb12-4923-9da4-c50cf77cc3c0">
If you debug, you can see that userConfig already has `mode: vue-scoped` before transformerDirectives are executed.